### PR TITLE
[#5650] improvement(web): create catalog page minor change

### PR DESCRIPTION
**Pull request by orenccl**
### What changes were proposed in this pull request?

- Capitalized display type values.
- Sorted the provider list alphabetically.

### Why are these changes needed?

Fix: #5650

### Does this PR introduce any user-facing changes?

No.

### How was this patch tested?

Opened the "Create Catalog" dialog and verified the changes.

### CODE REVIEW

**Comment by LauraXia123** [comment]
LGTM

**Review by LauraXia123 — APPROVED** [review]

File: web/web/src/lib/utils/initial.js [hunk 1]
```diff
@@ -48,6 +48,34 @@ export const messagingProviders = [
 ]
 
 export const providers = [
+  {
+    label: 'Apache Doris',
+    value: 'jdbc-doris',
+    defaultProps: [
+      {
+        key: 'jdbc-driver',
+        value: '',
```
Comment (tengqm) [inline web/web/src/lib/utils/initial.js:57]: Is there are default value for this driver?

File: web/web/src/lib/utils/initial.js [hunk 2]
```diff
@@ -48,6 +48,34 @@ export const messagingProviders = [
...
+        required: true,
+        description: 'e.g. com.mysql.jdbc.Driver'
+      },
+      {
+        key: 'jdbc-url',
+        value: '',
```
Comment (tengqm) [inline web/web/src/lib/utils/initial.js:63]: Why don't we put `jdbc:mysql://localhost:9030` as the default value?

File: web/web/src/lib/utils/initial.js (hunk 1 above)
Comment (orenccl) [reply web/web/src/lib/utils/initial.js:57]: yes

File: web/web/src/lib/utils/initial.js (hunk 2 above)
Comment (orenccl) [reply web/web/src/lib/utils/initial.js:63]: I think it's because, without a default value, the user becomes more aware of the value they are inputting.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**